app/controllers/subscriptions_controller.rb

```diff
@@ -27,6 +28,16 @@ def show
     authorize @subscription
   end
 
+  def activate
+    subscription = Subscription.find_by!(iugu_id: params[:data][:id])
+    authorize subscription
+    subscription.activate!
+    subscription.update!(
+      state: subscription.state,
```

Comment:
Our state machine already updated the state, so you don't need to update it here.

Reply:
When I tested, the `activate` was changing the object in memory, but it wasn't saving the changes to the DB. That's why I manually added the `update`. Is there another way to do this?

Reply:
Nevermind, I had something else wrong when I first tested it, now I removed the update and it worked :)